spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/GitRepo.java

```diff
@@ -236,7 +236,7 @@ private Ref checkoutBranch(File projectDir, String branch) throws GitAPIExceptio
 			return command.call();
 		}
 		catch (GitAPIException e) {
-			deleteBaseDirIfExists();
+			deleteBaseDirIfExists("Failed to delete base directory");
```

Comment:
Added the error message as a parameter as this is used when cloning and when checking out the repo so was throwing a very confusing error